src/coreclr/src/jit/flowgraph.cpp

```diff
@@ -25954,13 +25963,14 @@ PhaseStatus Compiler::fgTailMergeThrows()
     // We walk the map rather than the block list, to save a bit of time.
     BlockToBlockMap::KeyIterator iter(blockMap.Begin());
     BlockToBlockMap::KeyIterator end(blockMap.End());
-    int                          updateCount = 0;
+    unsigned                     updateCount = 0;
 
     for (; !iter.Equal(end); iter++)
     {
         BasicBlock* const nonCanonicalBlock = iter.Get();
         BasicBlock* const canonicalBlock    = iter.GetValue();
         flowList*         nextPredEdge      = nullptr;
+        bool              updated           = false;
```

Comment:
Why do we need this new local?

Reply:
Previously `updateCount` would overstate the number of cases transformed, so I added `updated` to get the accounting right.

Admittedly we don't use this now more accurate the count for much.  I had ambitions of using it to adjust `optNoReturnCallCount` so that the tail call heuristic in morph would sees a more accurate value and hence make better decisions, but doing that made things worse overall (bigger code size).

Reply:
I see, github shows diffs inaccurately, which that confused me and I didn't see that there was a semantic change. Previously we incremented `updateCount` once per predecessor and now we increment `updateCount` once per non-canonical block if there was a predecessor change. That makes sense.